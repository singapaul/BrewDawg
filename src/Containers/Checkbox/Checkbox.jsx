import React from 'react'

const Checkbox = ({label, checked, handleChange, id}) => {
  return (
    <>

    <label htmlFor={id}>{label}</label>
    <input aria-label={label} id={id} name={label} type="checkbox" checked={checked}  onChange={handleChange}/>
    <br />
    
    
    </>
  )
}

export default Checkbox