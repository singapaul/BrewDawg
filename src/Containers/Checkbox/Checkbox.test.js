import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import Checkbox from "./Checkbox";

// render returns object with a container
// Cheap and quick testing method
it("should match form component snapshot", () => {
  const { container } = render(
    <Checkbox label={"label"} checked={"true"} id={"1"} />
  );
  expect(container).toMatchSnapshot();
});
