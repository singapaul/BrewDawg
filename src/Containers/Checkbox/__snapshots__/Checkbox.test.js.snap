// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match form component snapshot 1`] = `
<div>
  <label
    for="1"
  >
    label
  </label>
  <input
    aria-label="label"
    checked=""
    id="1"
    name="label"
    type="checkbox"
  />
  <br />
</div>
`;
